lib/AST/Decl.cpp

```diff
@@ -1456,9 +1456,31 @@ PatternBindingDecl::create(ASTContext &Ctx, SourceLoc StaticLoc,
   auto PBE = PatternBindingEntry(Pat, EqualLoc, E, BindingInitContext);
   auto *Result = create(Ctx, StaticLoc, StaticSpelling, VarLoc, PBE, Parent);
 
-  if (BindingInitContext)
+  if (BindingInitContext) {
     cast<PatternBindingInitializer>(BindingInitContext)->setBinding(Result, 0);
 
+    // If the expression contains any closures, then we must change the
+    // closures' parents to `BindingInitContext`, because the closures are now
+    // children of `BindingInitContext`.
+    if (E) {
+      class Walker : public ASTWalker {
+      public:
+        DeclContext *NewParent;
+        Walker(DeclContext *NewParent) : NewParent(NewParent) {}
```

Comment:
Could you mark this `explicit` (even though its scoped tightly enough).  Did you try building on Windows?  It may require you to hoist this out of the scope.

Reply:
Marked explicit.

Seems like the windows build succeeded. There's some error in foundation tests, but that looks unrelated?